curvefs/src/client/fuse_s3_client.cpp

```diff
@@ -76,10 +74,270 @@ CURVEFS_ERROR FuseS3Client::Init(const FuseClientOption &option) {
+            // because the bolck Index is cumulative
+            uint64_t blockIndexEnd = blockIndexBegin + blockCounts - 1;
+
+            // the size of the last block
+            uint64_t lastBlockSize = leftSize % blockSize;
+            // whether the first block or the last block is full or not
+            bool firstBlockFull = (blockPos == 0) ? true : false;
+            bool lastBlockFull = (lastBlockSize == 0) ? true : false;
+            // the start and end block Index that need travel
+            uint64_t travelStartIndex, travelEndIndex;
+            // if the block is full, the size is needed download
+            // of the obj is blockSize. Otherwise, the value is special.
+            if (!firstBlockFull) {
+                travelStartIndex = blockIndexBegin + 1;
+                auto objectName = curvefs::common::s3util::GenObjName(
+                  chunkid, blockIndexBegin, compaction, fsId, ino);
+                prefetchObjs->push_back(std::make_pair(
+                  objectName, firstBlockSize));
+            } else {
+                travelStartIndex = blockIndexBegin;
+            }
+            if (!lastBlockFull) {
+                // block index is greater than or equal to 0
+                travelEndIndex = (blockIndexEnd == blockIndexBegin) ?
+                  blockIndexEnd : blockIndexEnd - 1;
+                auto objectName = curvefs::common::s3util::GenObjName(
+                  chunkid, blockIndexEnd, compaction, fsId, ino);
+                // there is no need to care about the order
+                // in which objects are downloaded
+                prefetchObjs->push_back(
+                  std::make_pair(objectName, lastBlockSize));
+            } else {
+                travelEndIndex = blockIndexEnd;
+            }
+            VLOG(9) << "travel obj, ino: " << ino
+                << ", chunkid: " << chunkid
+                << ", blockCounts: " << blockCounts
+                << ", compaction: " << compaction
+                << ", blockSize: " << blockSize
+                << ", chunkSize: " << chunkSize
+                << ", offset: "  << offset
+                << ", blockIndexBegin: " << blockIndexBegin
+                << ", blockIndexEnd: " << blockIndexEnd
+                << ", len: " << len
+                << ", firstBlockSize: " << firstBlockSize
+                << ", lastBlockSize: " << lastBlockSize
+                << ", blockPos: " << blockPos
+                << ", chunkPos: " << chunkPos;
+            for (auto blockIndex = travelStartIndex;
+              blockIndex <= travelEndIndex ; blockIndex++) {
+                auto objectName = curvefs::common::s3util::GenObjName(
+                    chunkid, blockIndex, compaction, fsId, ino);
+                prefetchObjs->push_back(std::make_pair(objectName, blockSize));
+            }
+        }
+    }
+}
+
+void FuseS3Client::WarmUpAllObjs(
+  const std::list<std::pair<std::string, uint64_t>> &prefetchObjs) {
```

Comment:
These logics are very similar to FileCacheManager::PrefetchS3Objs, try to merge as much as possible

Reply:
year, there are many similars, and i have added todo.